On Python ≥ 3.5, use [`pathlib.Path.mkdir`][1]:

    from pathlib import Path
    Path("/my/directory").mkdir(parents=True, exist_ok=True)

For older versions of Python, I see two answers with good qualities, each with a small flaw, so I will give my take on it:

Try [`os.path.exists`][2], and consider [`os.makedirs`][3] for the creation.

    import os
    if not os.path.exists(directory):
        os.makedirs(directory)

As noted in comments and elsewhere, there's a race condition &ndash; if the directory is created between the `os.path.exists` and the `os.makedirs` calls, the `os.makedirs` will fail with an `OSError`. Unfortunately, blanket-catching `OSError` and continuing is not foolproof, as it will ignore a failure to create the directory due to other factors, such as insufficient permissions, full disk, etc.

One option would be to trap the `OSError` and examine the embedded error code (see [Is there a cross-platform way of getting information from Python’s OSError][4]):

    import os, errno
    
    try:
        os.makedirs(directory)
    except OSError as e:
        if e.errno != errno.EEXIST:
            raise

Alternatively, there could be a second `os.path.exists`, but suppose another created the directory after the first check, then removed it before the second one &ndash; we could still be fooled. 

Depending on the application, the danger of concurrent operations may be more or less than the danger posed by other factors such as file permissions. The developer would have to know more about the particular application being developed and its expected environment before choosing an implementation.

Modern versions of Python improve this code quite a bit, both by exposing [`FileExistsError`][5] (in 3.3+)...

    try:
        os.makedirs("path/to/directory")
    except FileExistsError:
        # directory already exists
        pass

...and by allowing [a keyword argument to `os.makedirs` called `exist_ok`][6] (in 3.2+).

    os.makedirs("path/to/directory", exist_ok=True)  # succeeds even if directory exists.


  [1]: https://docs.python.org/library/pathlib.html#pathlib.Path.mkdir
  [2]: https://docs.python.org/library/os.path.html#os.path.exists
  [3]: https://docs.python.org/library/os.html#os.makedirs
  [4]: https://stackoverflow.com/questions/273698/is-there-a-cross-platform-way-of-getting-information-from-pythons-oserror
  [5]: https://docs.python.org/3.3/library/exceptions.html?#FileExistsError
  [6]: https://docs.python.org/3.2/library/os.html#os.makedirs